Integrated the static analysis type-checking tool flow

RyanChernoff opened the pull request:
**Context:** 
Because NodeBB is a large application, the current tests may not be enough to catch all bugs in our code. In order to catch more bugs or bugs that may occur later in development we are integrating the static analysis tool Flow. 

**Description:**
Flow is a type-checker for JavaScript which should help to ensure type safety in our code base. This should allow us to prevent simple type errors such as null differentiation. This tool should be run along side the linter during the dev process to ensure that all code in PRs are type safe. Unfortunately, Flow is prone to false negatives as it fully depends on marking files to evaluate and writing type annotations. However if a file is type annotated and marked at the top with '// @flow' then it should prove verry useful.

**File Changes:**
.babelrc
Necessary file for allowing flow to run correctly

.flowconfig
Config file for setting up flow

flowOutput
The output from the terminal of running flow on the application. **This is proof that it was run.**

flowInstructions.txt
A file with instruction on how to set up Flow locally via npm

jonassoh commented:
The PR is well formatted and looks good. There appears to be a fail in the lint test. It does not appear to be part of the code that you added or modified so we may have to look into why this fail has occurred now.

RyanChernoff commented:
> The PR is well formatted and looks good. There appears to be a fail in the lint test. It does not appear to be part of the code that you added or modified soma have to take a deeper dive to fix it.

It turns out my changes to install/package.json was causing the issue. I removed them and added a text file explaining the setup instead.

rmurugan58 commented:
The PR is well-formatted and detailed. All lint tests also pass now so everything looks good to me.

jonassoh reviewed (commented):
Code looks good and the instructions are clear
